background_scripts/completion_search.coffee

```diff
@@ -48,12 +48,19 @@ CompletionSearch =
     xhr = new XMLHttpRequest()
     xhr.open "GET", url, true
     xhr.timeout = 2500
-    xhr.ontimeout = xhr.onerror = -> callback null
-    xhr.send()
+    finished = 0
+    # Note(gdh1995): Maybe the 2 events below are not needed, but I just leave it unchanged for robustness
+    xhr.ontimeout = xhr.onerror = ->
+      finished++
+      if finished == 1
+        callback null
 
     xhr.onreadystatechange = ->
       if xhr.readyState == 4
-        callback if xhr.status == 200 then xhr else null
+        finished++
```

Comment:
Debatable, but I think this bookkeeping is clearer as a bool, because it explicitly communicates that we care only about calling the callback once. `finished` indicates that we care about how many times it was called, until you carefully read how it's used.

So something like this:

```
callbackCalled = false

xhr.ontimeout = ->
  if !callbackCalled
    callback null
  callbackCalled = true
```